### PR TITLE
Chore: Implementation

### PR DESCRIPTION
**Pull request by scotdalton**
@bloo WIP.

### CODE REVIEW

**Comment by bloo** [comment]
this is very cool @scotdalton

**Comment by scotdalton** [comment]
@bloo, yeah. i don't love that there are two different implementations of `ByteString` used (probuf and akka) but it's a necessary evil for now since generating the TMSP types from the protobuf file was dead easy. I also don't know how well it would play with another client's implementation, but that's a minor optimization.

**Comment by scotdalton** [comment]
@bloo and it needs some more tests 😃

**Comment by scotdalton** [comment]
@bloo, @armmimic last thing I want to do is get the tests running on CI, but this should be ready for review.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM velatrade/scala-base:latest
+
+COPY              build.sbt build.sbt
+COPY              project project
+COPY              src src
+COPY              test test
+
+RUN               sbt update
+RUN               sbt compile
```
Comment (bloo) [inline Dockerfile]: @scotdalton faster as one sbt run so JVM doesn't have to spin up

File: src/scala/tendermint/tmsp/Server.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  private val host = local.getHostString
+
+  private val port = local.getPort
+
+  private val processingFlow = Flow
+    .fromGraph(Framing.incoming)
+    .map(frame2request _)
+    .mapAsyncUnordered(parallelism) { request =>
+      // TODO: figure out how to actually make it async
```
Comment (bloo) [inline src/scala/tendermint/tmsp/Server.scala]: @scotdalton heh

File: Dockerfile (hunk 1 above)
Comment (scotdalton) [inline Dockerfile]: @bloo, good call!

File: src/scala/tendermint/tmsp/Server.scala (hunk 2 above)
Comment (scotdalton) [inline src/scala/tendermint/tmsp/Server.scala]: yeah, unfortunately I couldn't get the tests to actually complete without doing this. Not sure why. Definitely needs more investigation.

**Pull request merged**